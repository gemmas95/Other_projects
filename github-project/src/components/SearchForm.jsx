import React from "react";

function SearchForm(props) {
  return (
    <form
      onSubmit={props.onSubmit}
      data-testid="form"
      className="d-flex justify-content-center col-md-4 offset-md-4 flex-column text-center"
      autoComplete="off"
    >
      <label>Repository Name</label>
      <input
        className="form-group"
        placeholder="Repository name"
        name="repoName"
        value={props.dataRepo.repoName.trim()}
        onChange={props.onChange}
      ></input>

      <label>Owner</label>
      <input
        className="form-group"
        placeholder="Owner name"
        name="ownerName"
        type="text"
        value={props.dataRepo.ownerName.trim()}
        onChange={props.onChange}
      ></input>

      <button className="btn btn-info" data-testid="button" type="submit">
        View contributors!
      </button>
    </form>
  );
}

export default SearchForm;
