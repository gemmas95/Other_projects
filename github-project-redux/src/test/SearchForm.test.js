import renderer from "react-test-renderer";
import React from "react";
import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";
import { fireEvent } from "@testing-library/dom";

import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  let container = null;
  let dataRepo = { repoName: "tour-of-heroes", ownerName: "Bombasto" };
  const searchFormTree = renderer.create(<SearchForm dataRepo={dataRepo} />);
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it("should match snapshot", () => {
    expect(searchFormTree).toMatchSnapshot();
  });
  it("should call onSubmit and go to handleSubmit", () => {
    const handleSubmit = jest.fn((event) => event.preventDefault());
    const handleChange = jest.fn();

    act(() => {
      render(
        <SearchForm
          dataRepo={dataRepo}
          onChange={handleChange}
          onSubmit={handleSubmit}
        />,
        container
      );
      const submit = document.querySelector('[data-testid="form"]');

      fireEvent.submit(submit);
    });

    const button = document.querySelector('[data-testid="button"]');

    act(() => {
      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(handleSubmit).toHaveBeenCalled();
  });
});
