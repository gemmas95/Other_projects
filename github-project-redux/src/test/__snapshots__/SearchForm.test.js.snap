// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<form
  autoComplete="off"
  className="d-flex justify-content-center col-md-4 offset-md-4 flex-column text-center"
  data-testid="form"
>
  <label>
    Repository Name
  </label>
  <input
    className="form-group"
    name="repoName"
    placeholder="Repository name"
    value="tour-of-heroes"
  />
  <label>
    Owner
  </label>
  <input
    className="form-group"
    name="ownerName"
    placeholder="Owner name"
    type="text"
    value="Bombasto"
  />
  <button
    className="btn btn-info"
    data-testid="button"
    type="submit"
  >
    View contributors!
  </button>
</form>
`;
